egs/aishell/s10/chain/model.py

```diff
@@ -0,0 +1,148 @@
+  relu-batchnorm-layer name=tdnn1 dim=625
+  relu-batchnorm-layer name=tdnn2 input=Append(-1,0,1) dim=625
+  relu-batchnorm-layer name=tdnn3 input=Append(-1,0,1) dim=625
+  relu-batchnorm-layer name=tdnn4 input=Append(-3,0,3) dim=625
+  relu-batchnorm-layer name=tdnn5 input=Append(-3,0,3) dim=625
+  relu-batchnorm-layer name=tdnn6 input=Append(-3,0,3) dim=625
+
+  ## adding the layers for chain branch
+  relu-batchnorm-layer name=prefinal-chain input=tdnn6 dim=625 target-rms=0.5
+  output-layer name=output include-log-softmax=false dim=$num_targets max-change=1.5
+
+  # adding the layers for xent branch
+  # This block prints the configs for a separate output that will be
+  # trained with a cross-entropy objective in the 'chain' models... this
+  # has the effect of regularizing the hidden parts of the model.  we use
+  # 0.5 / args.xent_regularize as the learning rate factor- the factor of
+  # 0.5 / args.xent_regularize is suitable as it means the xent
+  # final-layer learns at a rate independent of the regularization
+  # constant; and the 0.5 was tuned so as to make the relative progress
+  # similar in the xent and regular final layers.
+  relu-batchnorm-layer name=prefinal-xent input=tdnn6 dim=625 target-rms=0.5
+  output-layer name=output-xent dim=$num_targets learning-rate-factor=$learning_rate_factor max-change=1.5
+'''
+
+
+# Create a network like the above one
+class ChainModel(nn.Module):
+
+    def __init__(feat_dim,
+                 output_dim,
+                 lda_mat_filename,
+                 hidden_dim=625,
+                 kernel_size_list=[1, 3, 3, 3, 3, 3],
+                 dilation_list=[1, 1, 1, 3, 3, 3],
+                 frame_subsampling_factor=3):
+        super().__init__()
+
+        assert len(kernel_size_list) == len(dilation_list)
+        num_layers = len(kernel_size_list)
+
+        tdnns = []
+        for i in range(num_layers):
+            in_channels = hidden_dim
+            if i == 0:
+                in_channels = feat_dim
+
+            kernel_size = kernel_size_list[i]
+            dilation = dilation_list[i]
+
+            # we do not need to perform padding in Conv1d because it
+            # has been included in left/right context while generating egs
+            layer = nn.Conv1d(in_channels=in_channels,
+                              out_channels=hidden_dim,
+                              kernel_size=kernel_size,
+                              dilation=dilation)
+            tdnns.append(layer)
+
+        self.tdnns = nn.ModuleList(tdnns)
+        self.batch_norms = nn.ModuleList([
+            nn.BatchNorm1d(num_features=hidden_dim) for i in range(num_layers)
+        ])
+
+        stride = frame_subsampling_factor
+
+        self.prefinal_chain_tdnn = nn.Conv1d(in_channels=hidden_dim,
+                                             out_channels=hidden_dim,
+                                             kernel_size=1,
+                                             stride=stride)
```

Comment:
@danpovey 
the subsampling is here.

This kind of layer
```
relu-batchnorm-layer name=tdnn4 input=Append(-3,0,3) dim=625
```
needs a dilation of `3`.